### PR TITLE
v0.65 updates round 4

### PR DESCRIPTION
**Pull request by turbocrime**
- JsonReadOptions are now more permissive and consistent throughout repo
- getAddressByIndex now handles missing inputs
- fetchers for same slightly adjusted

### CODE REVIEW

File: apps/webapp/src/fetchers/address.ts [hunk 1]
```diff
@@ -31,19 +24,16 @@ export const getAddresses = async (accounts: (number | undefined)[]): Promise<In
     }, {});
 };
 
-export const getAddressByIndex = async (account: number | undefined): Promise<Address> => {
-  const req = new AddressByIndexRequest();
-  if (account) req.addressIndex = new AddressIndex({ account });
-  const res = await viewClient.addressByIndex(req);
-  if (!res.address) throw new Error('Address not in getAddressByIndex response');
-  return res.address;
+export const getAddressByIndex = async (account = 0): Promise<Address> => {
```
Comment (grod220) [inline apps/webapp/src/fetchers/address.ts:27]: I go back and forth between whether we should have default account numbers for these functions. If someone is trying to get an address, they never should be passing nothing right?

**Review by grod220 — APPROVED** [review]
One comment, up to you

File: apps/webapp/src/fetchers/address.ts (hunk 1 above)
Comment (turbocrime) [reply apps/webapp/src/fetchers/address.ts:27]: well, *we* are sometimes passing nothing, from our own frontend. so

Comment (hdevalence) [reply apps/webapp/src/fetchers/address.ts:27]: imo passing nothing should be interpreted as the default account (index 0)

**Pull request merged**